core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -323,24 +324,17 @@ class ReplicaManager(val config: KafkaConfig,
       brokerTopicStats.removeMetrics(topic)
   }
 
-  def stopReplica(topicPartition: TopicPartition, deletePartition: Boolean)  = {
-    stateChangeLogger.trace(s"Handling stop replica (delete=$deletePartition) for partition $topicPartition")
-
+  def stopReplica(topicPartition: TopicPartition, deletePartition: Boolean): Unit  = {
     if (deletePartition) {
       getPartition(topicPartition) match {
```

Comment:
Could potentially use `nonOfflinePartition(topicPartition).foreach`

Reply:
Actually, it does not work because we need both the reference to the hosted partition and the partition bellow: `hostedPartition` and `removedPartition`. `nonOfflinePartition` only provides the latter.